test/e2e/antreapolicy_test.go

```diff
@@ -427,6 +427,46 @@ func testInvalidACNPAppliedToNotSetInAllRules(t *testing.T) {
 	}
 }
 
+func testInvalidACNPAppliedToGroupSetWithPodSelector(t *testing.T) {
+	invalidNpErr := fmt.Errorf("invalid Antrea ClusterNetworkPolicy with appliedTo set group with podSelector")
```

Comment:
maybe we should consider moving some e2e tests to unit tests if they don't require a whole system to validate the code. For example, the illegal inputs that can be validated by calling the validation function directly. This can enable greater converage with a table driven tests and reduce resources spent on e2e tests.